nmdc_schema/metab_id_refactor.py

```diff
@@ -0,0 +1,833 @@
+                        del(metab_activities_doc["has_metabolite_quantifications"])
+                        activity_without_omics_processing.append(metab_activities_doc)
+
+                    for omics_processing_doc in omics_processing_docs:
+                        
+                        new_raw_file_id = mint_nmdc_id(NMDC_Types.DataObject, 1)[0]
+
+                        omics_processing_update = { "$set": { "has_output": [new_raw_file_id]} }    
+
+                        omics_processing_set.update_one(omics_processing_doc, omics_processing_update)
+
+                        new_omics_processing_id = omics_processing_doc['id']
+
+                        new_data_product_id = mint_nmdc_id(NMDC_Types.DataObject, 1)[0]
+
+                        metab_new_id = mint_nmdc_id(NMDC_Types.MetabolomicsAnalysisActivity, 1)[0]
+
+                        update_has_calibration_object(metab_activities_doc.get('has_calibration'))
+
+                        update_data_products(new_data_product_id, new_raw_file_id, omics_processing_doc.get("has_output"), metab_activities_doc, new_omics_processing_id, metab_new_id)
+                        
+                        metab_activity_update = { "$set": { "id": metab_new_id , "has_output":[new_data_product_id],
+                                                "has_input":[new_raw_file_id], "was_informed_by": new_omics_processing_id} }
+
+                        
+                        activity_with_omics_processing.append(metab_activities_doc)
+                        omics_processing_with_activities.append(omics_processing_doc)
+                        metab_activities_set.update_one(metab_activities_doc, metab_activity_update)
+    
+    json.dump(activity_with_omics_processing, open('./metab_analysis_activity_with_omics_porcessing.json', 'w'), cls=JSONEncoder)
+    json.dump(omics_processing_with_activities, open('./metab_omics_porcessing_with_analysis_activity.json', 'w'), cls=JSONEncoder)
+    json.dump(activity_without_omics_processing, open('./metab_analysis_activity_without_omics_porcessing.json', 'w'), cls=JSONEncoder)
+
+    omics_processing_to_delete = [] 
+    for omics_processing_doc in omics_processing_set.find({}):
+        
+        #print(nom_activities_doc)
+        has_outputs = omics_processing_doc.get("has_output")
+        if has_outputs:
+            if omics_processing_doc.get("processing_institution") == 'EMSL':
+                for has_output in has_outputs:
+                    if has_output.startswith('emsl'):
+                        if omics_processing_doc['omics_type']['has_raw_value'] != 'Lipidomics':
+                            #pprint(omics_processing_doc)     
+                            omics_processing_to_delete.append(omics_processing_doc)                   
+
+    json.dump(omics_processing_to_delete, open('./metab_omics_porcessing_without_analysis_activity.json', 'w'), cls=JSONEncoder)
+
+def update_nom_omics_processing(raw_file_path=None):
```

Comment:
Deletion calls are commented out, does this need to be uncommented for production.